### PR TITLE
Analyze and test more cheatsheet snippets

### LINKED ISSUE #2295: Add and auto-include cheatsheet code samples
We need to add the cheatsheet code samples in the repo so that we can link the code-excerpts in the source file.

Context: #1803 

cc: @kwalrath

### PR DESCRIPTION
**Pull request by parlough**
Fixes #2295

### CODE REVIEW

**Review by kwalrath — COMMENTED** [review]
LGTM! We should also get an engineer to review the code.

@domesticmouse? @johnpryan?

File: examples/misc/test/cheatsheet/arguments_test.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    as optional_args;
+import 'package:examples/cheatsheet/optional_positional_args2.dart'
+    as defaulted_args;
+import 'package:test/test.dart';
+
+void main() {
+  test('print_name', () {
+    expect(() {
+      printNameTest();
+    }, prints('Avinash Gupta \nPoshmeister Moneybuckets IV\n'));
```
Comment (johnpryan) [inline examples/misc/test/cheatsheet/arguments_test.dart:12]: nit: I realize this is testing an already existing snippet, but this string might be too wacky / goofy: https://developers.google.com/style/tone

**Review by johnpryan — APPROVED** [review]

File: examples/misc/test/cheatsheet/arguments_test.dart (hunk 1 above)
Comment (parlough) [reply examples/misc/test/cheatsheet/arguments_test.dart:12]: That's fair, we don't want the examples to introduce unnecessary confusion or distraction. There's a few examples like this and I'm not sure of the best replacement at the moment, so how about we move this to a separate, larger issue to consider the voice/tone/global understanding of various examples. If not, do you have any short term suggestions for more understandable arguments I could switch these for.

Comment (johnpryan) [reply examples/misc/test/cheatsheet/arguments_test.dart:12]: Sounds good, let's consider this separately https://github.com/dart-lang/site-www/issues/3485

**Pull request merged**